src/pages/patientView/pathwayMapper/PatientViewPathwayMapper.tsx

```diff
@@ -0,0 +1,291 @@
+    }
+    @computed get alterationFrequencyDataForQueryGenes() {
+        this.props.store.mergedMutationDataIncludingUncalledFilteredByGene.forEach(
+            altData => {
+                const mutationType = {
+                    gene: altData[0].gene.hugoGeneSymbol,
+                    altered: 1,
+                    sequenced: 1,
+                    percentAltered: altData[0].mutationType,
+                    geneticTrackRuleSetParams: DEFAULT_RULESET_PARAMS,
+                    geneticTrackData: this.props.store.geneticTrackData.result
+                        ? this.props.store.geneticTrackData.result[
+                              altData[0].gene.hugoGeneSymbol
+                          ]
+                        : undefined,
+                };
+                if (mutationType) {
+                    alterationFrequencyData.push(mutationType);
+                }
+            }
+        );
+
+        this.props.store.mergedDiscreteCNADataFilteredByGene.forEach(
+            altData => {
+                const cna = {
+                    gene: altData[0].gene.hugoGeneSymbol,
+                    altered: 1,
+                    sequenced: 1,
+                    percentAltered: this.getCNAtypes(altData[0].alteration),
+                    geneticTrackRuleSetParams: DEFAULT_RULESET_PARAMS,
+                    geneticTrackData: this.props.store.geneticTrackData.result
+                        ? this.props.store.geneticTrackData.result[
+                              altData[0].gene.hugoGeneSymbol
+                          ]
+                        : undefined,
+                };
+                if (cna) {
+                    alterationFrequencyData.push(cna);
+                }
+            }
+        );
+
+        return alterationFrequencyData;
+    }
+    private getCNAtypes(CNAtype: number) {
+        if (CNAtype === 1) return 'GAIN';
+        else if (CNAtype === 0) return 'DIPLOID';
+        else if (CNAtype === -1) return 'SHALLOWDEL';
+        else if (CNAtype === -2) return 'DeepDel';
+        else return 'AMP';
+    }
+
+    private getQueryGenes(data: ICBioData[]) {
+        const allTypes = data.map(x => x.gene);
+
+        const allGenes = allTypes.filter((x, i, a) => a.indexOf(x) == i);
+        //This parameter needs the hugoGeneSymbol in PathwayMapper
+        const keyed_genes = allGenes.map(gene => {
+            return { hugoGeneSymbol: gene };
+        });
+        return keyed_genes;
+    }
+    @computed get isNewStoreReady() {
+        return (
+            this.storeForAllData &&
+            this.storeForAllData.samples.isComplete &&
+            this.storeForAllData.mergedMutationData &&
+            this.storeForAllData.coverageInformation.isComplete &&
+            this.storeForAllData.mergedDiscreteCNADataFilteredByGene &&
+            this.storeForAllData
+                .mergedMutationDataIncludingUncalledFilteredByGene
+        );
+    }
```

Comment:
This is a bit confusing, since we only have one store here. If we still need to check if data is complete within the component, then let's rename this function.